lightkurve/correctors.py

```diff
@@ -487,3 +491,212 @@ def breakpoints(self, campaign):
+        """
+
+        # parse aperture
+        aperture = self.tpf._parse_aperture_mask(aperture_mask)
+
+        # find pixel bounds of aperture on tpf
+        xmin, xmax = min(np.where(aperture)[0]),  max(np.where(aperture)[0])
+        ymin, ymax = min(np.where(aperture)[1]),  max(np.where(aperture)[1])
+
+        # crop data cube to include only desired pixels
+        # this is required for superstamps to ensure matrix is invertable
+        flux_crop = self.flux[:, xmin:xmax+2, ymin:ymax+2]
+        flux_err_crop = self.flux_err[:, xmin:xmax+2, ymin:ymax+2]
+        aperture_mask = aperture[xmin:xmax+2, ymin:ymax+2]
+
+        # set transit mask
+        if len(transit_mask) == 0:
+            transit_mask = np.where(self.time)
+        M = lambda x: x[transit_mask]
+
+        #  generate flux light curve from desired pixels
+        lc = M(self.tpf).to_lightcurve(aperture_mask=aperture_mask)
+
+        # set aperture values
+        aperture_vals = np.zeros(aperture_mask.shape)
+        aperture_vals[np.where(aperture_mask)] = 1
+
+        # compute raw flux light curve
+        self.aperture_flux = np.array([f*aperture_vals for f in flux_crop]).reshape(len(flux_crop), -1)
+        rawflux = np.sum(self.aperture_flux.reshape(len(self.aperture_flux), -1), axis=1)
+
+        # first order PLD
+        f1 = self.aperture_flux / rawflux.reshape(-1, 1)
+        pca = PCA(n_components=n_components_first)
+        X1 = pca.fit_transform(f1)
+
+        # second order PLD
+        f2 = np.product(list(multichoose(f1.T, 2)), axis = 1).T
+        pca = PCA(n_components=n_components_second)
+        X2 = pca.fit_transform(f2)
+
+        # combine them and add a column vector of 1s for stability
+        X = np.hstack([np.ones(X1.shape[0]).reshape(-1, 1), X1, X2])
+
+        # mask transits in design matrix
+        MX = M(X)
+
+        # define gaussian process parameters amp, tau
+        y = M(rawflux) - np.dot(MX, np.linalg.solve(np.dot(MX.T, MX),
+                                np.dot(MX.T, M(rawflux))))
+        amp = np.nanstd(y)
+        tau = 30.
+
+        # set up gaussian process
+        kernel = celerite.terms.Matern32Term(np.log(amp), np.log(tau))
+        gp = celerite.GP(kernel)
+        sigma = gp.get_matrix(M(self.time)) + \
+                np.diag(np.sum(M(flux_err_crop).reshape(len(M(flux_err_crop)),
+                        -1), axis=1)**2)
+
+        # compute
+        A = np.dot(MX.T, np.linalg.solve(sigma, MX))
+        B = np.dot(MX.T, np.linalg.solve(sigma, M(rawflux)))
+        C = np.linalg.solve(A, B)
+
+        # compute detrended light curve
+        model = np.dot(X, C)
+        self.detrended_flux = rawflux - model + np.nanmean(rawflux)
+
+        # calculate errors (ignore warnings related to zero or negative errors)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", RuntimeWarning)
+            flux_err = np.nansum(self.flux_err[:, aperture_mask]**2, axis=1)**0.5
+
```

Comment:
If I understand this correctly, at line 634 ish you're calculating the maximum likelihood estimation of `tau` and `amp`. But not computing the errors on those parameters. If we were to calculate the errors on those parameters, we could propagate the error on the model, and then propagate them through to `flux_err`.

This might be a functionality that would make things rather slow, so it might be worth having a switch to turn something like this off.

Reply:
This is a good suggestion, and is likely more within the scope of a standalone GP fitter in `lightkurve`. I'll include it when we add the new GP function.